src/test/typescript/typescript-preparser_test.ts

```diff
@@ -48,6 +52,37 @@ suite('TypeScriptParser', () => {
           sourceFile.statements[0].kind, ts.SyntaxKind.ImportDeclaration);
     });
 
+    test('throws a WarningCarryingException for parse errors', async() => {
+      const contents = 'const const const const const #!@(~~)!();';
+      const url = 'ts-parse-error.ts';
+      let error: WarningCarryingException|undefined = undefined;
+      try {
+        parser.parse(contents, url);
+      } catch (e) {
+        if (!(e instanceof WarningCarryingException)) {
+          console.log(e);
+          throw new Error('Expected a warning carrying exception.');
+        }
+        error = e;
+      }
+      if (error === undefined) {
+        throw new Error('Parsing invalid file did not throw!');
+      }
+      const warningPrinter = new WarningPrinter(null as any, {
+        analyzer:
+            new Analyzer({urlLoader: new TestUrlLoader({[url]: contents})})
+      });
+
+      async function getUnderlinedText(warning: Warning) {
```

Comment:
There's a lot of ceremony here and elsewhere doing this same thing. I'm pulling the common code out to a class in test utils in a followup